core/src/credential.rs

```diff
@@ -75,6 +76,102 @@ pub struct Credential {
     pub property_set: std::collections::HashMap<String, serde_json::Value>,
 }
 
+#[derive(Debug, serde::Serialize, serde::Deserialize, Clone, Validate)]
+pub struct VerifiableCredential {
+    #[serde(flatten)]
+    #[validate]
+    pub credential: Credential,
+    pub proof: crate::proof::CredentialProof,
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize, Clone, Validate)]
+pub struct Presentation {
+    #[serde(rename = "@context")]
+    pub context: DocumentContext,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub id: Option<String>,
+
+    #[serde(rename = "type")]
+    pub presentation_type: Vec<PresentationType>,
+
+    #[serde(rename = "verifiableCredential")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[validate]
+    pub verifiable_credential: Option<Vec<VerifiableCredential>>,
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize, Clone, Validate)]
+pub struct VerifiablePresentation {
+    #[serde(flatten)]
+    #[validate]
+    pub presentation: Presentation,
+
+    pub proof: crate::proof::CredentialProof,
+}
+
+impl FromStr for CredentialType {
+    type Err = super::error::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "VerifiableCredential" => Ok(CredentialType::VerifiableCredential),
+            "PermanentResidentCard" => Ok(CredentialType::PermanentResidentCard),
+            "BankCard" => Ok(CredentialType::BankCard),
+            "BankAccount" => Ok(CredentialType::BankAccount),
+            "UniversityDegreeCredential" => Ok(CredentialType::UniversityDegreeCredential),
+            "AlumniCredential" => Ok(CredentialType::AlumniCredential),
+            _ => Err(super::error::Error::Unknown(
+                "Unknown CredentialType".to_string(),
+            )),
+        }
+    }
+}
+
+impl CredentialType {
+    pub fn as_str(&self) -> &str {
+        match self {
+            CredentialType::VerifiableCredential => "VerifiableCredential",
+            CredentialType::PermanentResidentCard => "PermanentResidentCard",
+            CredentialType::BankCard => "BankCard",
+            CredentialType::BankAccount => "BankAccount",
+            CredentialType::UniversityDegreeCredential => "UniversityDegreeCredential",
+            CredentialType::AlumniCredential => "AlumniCredential",
+        }
+    }
+}
+
+impl std::fmt::Display for VerifiableCredential {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self.to_json_string() {
+            Ok(vc) => {
+                write!(f, "{}", vc)
+            }
+            Err(e) => {
+                write!(f, "Error: {}", e)
+            }
+        }
+    }
+}
+
+impl FromStr for Credential {
+    type Err = super::error::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let c = Credential::from_json_str(s)?;
+        Ok(c)
```

Comment:
This can just be a bare `Credential::from_json_str`, and if this repo had clippy configured the same way that `core` does I think it would emit a style warning for writing it this way